lib/timeutils/tests/test_unixtime.c

```diff
@@ -73,6 +75,347 @@ Test(unixtime, unix_time_fix_timezone_adjusts_timestamp_as_if_was_parsed_assumin
+   * of tests as we go through the daylight saving start hour in the spring.
+   * */
+
+#define TESTCASE_START(base_ofs) \
+  do { \
+    ut = base_ut; \
+    ut.ut_sec += base_ofs; \
+  } while (0)
+
+  /* Base timestamp: Mar 10 2019 02:00:00 CET,
+   *
+   * e.g.  if interpreted in EST5DST this is the start of the DST transition
+   * hour in 2019.  We simulate that we assumed it was CET and then we "fix"
+   * the timezone to EST5EDT */
+
+  base_ut.ut_sec = 1552179600;
+  base_ut.ut_gmtoff = 3600;
+
+  /* testcase, 1 second earlier than the DST transition hour */
+
+  TESTCASE_START(-1);
+
+  /* this "fix" will cause ut->ut_sec to be changed, it will go forward, but
+   * still not be reaching the daylight saving start, short of a second */
+
+  unix_time_fix_timezone_with_tzinfo(&ut, cached_get_time_zone_info("EST5EDT"));
+
+  /* still has not reached the DST start time */
+  cr_assert(ut.ut_sec == 1552201200 - 1);
+  /* thus the resulting timezone is still EST and not the daylight saving variant */
+  cr_assert(ut.ut_gmtoff == -5*3600);
+
+  /* next testcase, 1 second later, e.g.  Mar 10 2019 02:00:00 CET, which is
+   * exactly the daylight saving start second.  It should be converted to
+   * Mar 10 2019 03:00:00 EDT */
+
+  TESTCASE_START(0);
+
+  unix_time_fix_timezone_with_tzinfo(&ut, cached_get_time_zone_info("EST5EDT"));
+
+  /* we are at exactly the the DST start time */
+  cr_assert(ut.ut_sec == 1552201200);
+  /* thus the resulting timezone is EDT and not EST */
+  cr_assert(ut.ut_gmtoff == -4*3600);
+
+  /* next testcase, 30 minutes later, e.g. 02:30:00, that is converted to 03:30:00 */
+  TESTCASE_START(1800);
+
+  unix_time_fix_timezone_with_tzinfo(&ut, cached_get_time_zone_info("EST5EDT"));
+
+  /* we lose the hour, so the DST start time */
+  cr_assert(ut.ut_sec == 1552201200 + 1800);
+  /* thus the resulting timezone is EDT */
+  cr_assert(ut.ut_gmtoff == -4*3600);
+
+  /* next testcase, 1 hour second later, e.g. 03:00:00 */
+  TESTCASE_START(3600);
+
+  unix_time_fix_timezone_with_tzinfo(&ut, cached_get_time_zone_info("EST5EDT"));
+
+  /* we lose the hour, so the DST start time */
+  cr_assert(ut.ut_sec == 1552201200);
+  /* thus the resulting timezone is still EST and not the daylight saving variant */
+  cr_assert(ut.ut_gmtoff == -4*3600);
+
+  /* next testcase, 2 hours second later, e.g. 04:00:00 */
+  TESTCASE_START(7200);
+
+  unix_time_fix_timezone_with_tzinfo(&ut, cached_get_time_zone_info("EST5EDT"));
+
+  /* we lose the hour, so the DST start time */
+  cr_assert(ut.ut_sec == 1552201200 + 3600);
+  /* thus the resulting timezone is still EST and not the daylight saving variant */
+  cr_assert(ut.ut_gmtoff == -4*3600);
+
+#undef TESTCASE_START
+}
+
+Test(unixtime, unix_time_fix_timezone_with_tzinfo_to_a_zone_forwards_during_sprint_daylight_saving_hour)
+{
+  UnixTime ut = UNIX_TIME_INIT;
+  UnixTime base_ut;
+
+
+#define TESTCASE_START(base_ofs) \
```

Comment:
Me seems as if this is repetitively defined in each test. Would it make sense to pull out globally?

Reply:
I have added a commit that replaces the macro with a local function. I originally wanted to "highlight" each step in that testcase function and I've figured a macro with all caps name would be a good choice.

I moved that into the comment right above the function call I just introduced.